File: fe/lib/models/catalog.dart
class Catalog {
  final String catalogId;
  final String userId;
  final String title;
  final String description;
  final String category;
  final List<String> tags;
  final String visibility;
  final String? thumbnailUrl;
  final DateTime createdAt;
  final DateTime updatedAt;
  final int itemCount;
  final int ownedCount;
  final double completionRate;
  final String? originalCatalogId; // 복사본인 경우 원본 카탈로그 ID

  Catalog({
    required this.catalogId,
    required this.userId,
    required this.title,
    required this.description,
    required this.category,
    required this.tags,
    required this.visibility,
    this.thumbnailUrl,
    required this.createdAt,
    required this.updatedAt,
    this.itemCount = 0,
    this.ownedCount = 0,
    this.completionRate = 0.0,
    this.originalCatalogId,
  });

  factory Catalog.fromJson(Map<String, dynamic> json) {
    return Catalog(
      catalogId: json['catalog_id'] as String,
      userId: json['user_id'] as String,
      title: json['title'] as String,
      description: json['description'] as String,
      category: json['category'] as String,
      tags:
          (json['tags'] as List<dynamic>?)?.map((e) => e.toString()).toList() ??
              [],
      visibility: json['visibility'] as String,
      thumbnailUrl: json['thumbnail_url'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
      itemCount: json['item_count'] as int? ?? 0,
      ownedCount: json['owned_count'] as int? ?? 0,
      completionRate: (json['completion_rate'] as num?)?.toDouble() ?? 0.0,
      originalCatalogId: json['original_catalog_id'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'catalog_id': catalogId,
      'user_id': userId,
      'title': title,
      'description': description,
      'category': category,
      'tags': tags,
      'visibility': visibility,
      'thumbnail_url': thumbnailUrl,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
      'item_count': itemCount,
      'owned_count': ownedCount,
      'completion_rate': completionRate,
      'original_catalog_id': originalCatalogId,
    };
  }
}

File: be/catalog-api/app/schemas/catalog.py
from pydantic import BaseModel, Field
from typing import List, Optional

class CatalogBase(BaseModel):
    title: str = Field(..., description="카탈로그 제목")
    description: str = Field(..., description="카탈로그 설명")
    category: str = Field(default="미분류", description="카테고리")
    tags: List[str] = Field(default_factory=list, description="태그 리스트")
    visibility: str = Field(default="public", description="공개 여부 (public/private)")
    thumbnail_url: Optional[str] = Field(default=None, description="썸네일 이미지 URL")

class CatalogCreate(CatalogBase):
    pass

class CatalogUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    category: Optional[str] = None
    tags: Optional[List[str]] = None
    visibility: Optional[str] = None
    thumbnail_url: Optional[str] = None

class Catalog(CatalogBase):
    catalog_id: str = Field(..., description="카탈로그 고유 ID")
    user_id: str = Field(..., description="소유자 ID")
    created_at: str = Field(..., description="생성일")
    updated_at: str = Field(..., description="수정일")
    item_count: Optional[int] = Field(default=0, description="아이템 개수")
    owned_count: Optional[int] = Field(default=0, description="보유 아이템 개수")
    completion_rate: Optional[float] = Field(default=0.0, description="수집률")
    original_catalog_id: Optional[str] = Field(default=None, description="원본 카탈로그 ID (복사본인 경우)")
    creator_nickname: Optional[str] = Field(default=None, description="생성자 닉네임 (공개 카탈로그 조회 시)")
    is_saved: Optional[bool] = Field(default=False, description="저장 여부 (공개 카탈로그 조회 시)")

    class Config:
        from_attributes = True
